File: TyTraCLTransformations/MemoryReduction/test_pass_array_slice.f90
! $gfortran-mp-10 -L/usr/lib test_pass_array_slice.f90
program testPassArraySlice
integer, dimension(4,4) :: v_s_i
integer, dimension(4) :: v_o
integer :: i,j, n

    do i=1,4
    do j=1,4
        v_s_i(i,j)=i-1+4*(j-1)
!        print *,v_s_i(i,j)a
    end do
        print *,v_s_i(i,:)
    end do
    
    n=1

    do i=1,4
      call f(v_s_i(i,:),v_o(i))
    end do


    print *, v_o;
end program testPassArraySlice

subroutine f(x_in, x_out) 
    integer, dimension(4) :: x_in
    integer :: x_out, res, i
    res=0
    do i=1,4
         res = res +  x_in(i)
    end do
    x_out = res;
end subroutine f 

File: TyTraCLTransformations/MemoryReduction/test_pass_array_slice.c
#include <stdio.h>

void f(int[4],int*);

int main() {
int v_s_i[4][4];
int v_o[4];
int i,j;

    int  n;
    for (i=0;i<4;++i) {
    for (j=0;j<4;++j) {
        v_s_i[i][j]=i+4*j;
        printf("%d ",v_s_i[i][j]);
    }
    printf("\n");
    }
    n=1;


    for (i=0;i<4;++i) {
      f(v_s_i[i],&v_o[i]);
    }



    for (i=0;i<4;++i) {
        printf("%d\n",v_o[i]);
    }
    return 0;
}

void f(int x_in[4],int* x_out) {
    int res=0;
    for (int i=0;i<4;++i) {
         res +=   x_in[i];// * x_in[i];
    }
    *x_out = res;
}    
